src/analysis/convex/cone.lean

```diff
@@ -272,29 +326,30 @@ def to_ordered_add_comm_group (S : convex_cone E) (h₁ : pointed S) (h₂ : sal
   ..to_partial_order S h₁ h₂,
   ..show add_comm_group E, by apply_instance }
 
+end add_comm_group
+end ordered_semiring
+
 /-! ### Positive cone of an ordered module -/
-section positive_cone
 
-variables (M : Type*) [ordered_add_comm_group M] [module ℝ M] [ordered_smul ℝ M]
+section positive_cone
+variables (𝕜 E) [ordered_semiring 𝕜] [ordered_add_comm_group E] [module 𝕜 E] [ordered_smul 𝕜 E]
 
 /--
 The positive cone is the convex cone formed by the set of nonnegative elements in an ordered
 module.
 -/
-def positive_cone : convex_cone M :=
+def positive_cone : convex_cone 𝕜 E :=
 { carrier := {x | 0 ≤ x},
   smul_mem' :=
     begin
-      intros c hc x hx,
-      have := smul_le_smul_of_nonneg (show 0 ≤ x, by exact hx) (le_of_lt hc),
-      have h' : c • (0 : M) = 0,
-      { simp only [smul_zero] },
-      rwa [h'] at this
+      rintro c hc x (hx : _ ≤ _),
+      rw ←smul_zero c,
+      exact smul_le_smul_of_nonneg hx hc.le,
     end,
-  add_mem' := λ x hx y hy, add_nonneg (show 0 ≤ x, by exact hx) (show 0 ≤ y, by exact hy) }
+  add_mem' := λ x (hx : _ ≤ _) y (hy : _ ≤ _), add_nonneg hx hy }
 
 /-- The positive cone of an ordered module is always salient. -/
-lemma salient_of_positive_cone : salient (positive_cone M) :=
+lemma salient_positive_cone : salient (positive_cone 𝕜 E) :=
```

Comment:
While we're here, would you mind replacing the `tactic.ring.XXX` lemma used in the proof with:
`by rw [(sub_eq_add_neg x x).symm]; exact sub_self x,`?

Reply:
Golfed it hard ⛳